sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/WindowedSubscriber.java

```diff
@@ -0,0 +1,1157 @@
+                        // Begin: releaser-loop
+                        while (released != r) {
+                            final boolean workArrived = !workQueue.isEmpty();
+                            if (workArrived) {
+                                break;
+                            }
+                            final T item = queue.poll();
+                            if (item == null || isDoneOrCancelled()) {
+                                break;
+                            }
+                            try {
+                                releaser.accept(item);
+                            } catch (Throwable e) {
+                                this.logger.atError().log("Unexpected: 'releaser' thrown error.", e);
+                            }
+                            released++;
+                        }
+                        // End: releaser-loop
+
+                        if (released != 0 && r != Long.MAX_VALUE) {
+                            REQUESTED.addAndGet(this, -released);
+                        }
+                    }
+                }
+            }
+
+            missed = WIP.addAndGet(this, -missed);
+            if (missed == 0) {
+                break;
+            }
+            // The next serialized drain-loop iteration to process missed signals arrived during last iteration.
+        }
+        // End: drain-loop
+    }
+
+    /**
+     * Check if upstream terminated (with error or completion) or downstream canceled the subscription.
+     *
+     * @return true if this subscriber is terminated.
+     */
+    private boolean isDoneOrCancelled() {
+        return done || s == Operators.cancelledSubscription();
+    }
+
+    /**
+     * CONTRACT: Never invoke from the outside of serialized drain-loop.
+     * <p>
+     * initialize the work and request the items needed to meet its window demand.
+     * </p>
+     *
+     * @param w the work.
+     */
+    private void initWorkOnce(WindowWork<T> w) {
+        if (!w.init()) {
+            return;
+        }
+
+        // 'requested' is the overall outstanding demand yet to be satisfied by the upstream.
+        final long requested = REQUESTED.get(this);
+        final long workDemand = w.getDemand();
+        final long difference = workDemand - requested;
+
+        final LoggingEventBuilder logger = w.getLogger()
+            .addKeyValue(UPSTREAM_REQUESTED_KEY, requested)
+            .addKeyValue(DIFFERENCE_KEY, difference);
+
+        if (difference > 0) {
+            Operators.addCap(REQUESTED, this, difference);
+            logger.log("Initialized: request-upstream:true.");
+            s.request(difference);
+        } else {
+            logger.log("Initialized: request-upstream:false.");
+        }
+    }
+
+    /**
+     * Post a timed-out or canceled work to the drain-loop to process.
+     *
+     * @param w canceled or timed-out work.
+     */
+    private void postTimedOutOrCanceledWork(WindowWork<T> w) {
```

Comment:
nit: `canceled` is used here but in line 424 below, `cancelled` is used.

Reply:
thanks, made it consistent.